src/library_nodefs.js

```diff
@@ -292,6 +292,27 @@ mergeInto(LibraryManager.library, {
         }
 
         return position;
+      },
+      mmap: function(stream, buffer, offset, length, position, prot, flags) {
+        if (!FS.isFile(stream.node.mode)) {
+          throw new FS.ErrnoError({{{ cDefine('ENODEV') }}});
+        }
+        var ptr = _malloc(length);
+        NODEFS.stream_ops.read(stream, buffer, ptr + offset , length, position);
+        
```

Comment:
comparing to MEMFS, `offset` is not used there, but you add it here - is that a bug there, or here? If we aren't sure, an assertion that the offset is 0 might be good enough for now.

Reply:
not sure actually, so I've added the assertion as you suggest.